src/ceph-volume/ceph_volume/devices/lvm/batch.py

```diff
@@ -436,10 +439,20 @@ def get_deployment_layout(self) -> List["OSD"]:
                 len(very_fast_allocations), num_osds))
             exit(1)
 
-        for osd in plan:
-            if fast_devices:
-                osd.add_fast_device(*fast_allocations.pop(),
-                                    type_=fast_type)
+        if fast_devices:
+            fast_alloc: Optional[tuple[str, float, disk.Size, int]] = None
+            for osd in plan:
+                if self.args.block_db_size is not None and not self.args.db_devices:
+                    for i, fast_alloc in enumerate(fast_allocations):
+                        if osd.data.path == fast_alloc[0]:
+                            fast_alloc = fast_allocations.pop(i)
```

Comment:
Is the `fast_alloc` being looped over in `enumerate(fast_allocations)` and the `fast_alloc` being set in `fast_alloc = fast_allocations.pop(i)` the same thing? Are we just setting it again to make sure it's taken out of `fast_allocations`?

Reply:
correct, although it has no impact, the `fast_alloc` being looped over should have a different name to avoid confusion, didn't notice that, good catch 👍